File: packages/backend/src/features/question/mapper/question.mapper.ts
import { AnswerDto } from 'src/features/question/api/dto/answer.dto';
import { QuestionDto } from 'src/features/question/api/dto/question.dto';
import { AnswerModel } from 'src/features/question/model/answer.model';
import { QuestionModel } from 'src/features/question/model/question.model';

export class QuestionMapper {
  static toQuestionDto(model: QuestionModel): QuestionDto {
    return {
      id: model.id,
      createdAt: model.createdAt,
      updatedAt: model.updatedAt,
      text: model.text,
      mediaType: model.mediaType,
      answerMode: model.answerMode,
      answers: model.answers.map((answer) => this.toAnswerDto(answer)),
    };
  }

  static toAnswerDto(model: AnswerModel): AnswerDto {
    return {
      id: model.id,
      createdAt: model.createdAt,
      updatedAt: model.updatedAt,
      text: model.text,
      isCorrectAnswer: model.isCorrectAnswer,
    };
  }
}

File: packages/backend3/src/main/kotlin/quizzem/backend/features/question/mapper/QuestionMapper.kt
package quizzem.backend.features.question.mapper

import quizzem.backend.features.question.api.dto.AnswerDto
import quizzem.backend.features.question.api.dto.QuestionDto
import quizzem.backend.features.question.model.AnswerModel
import quizzem.backend.features.question.model.QuestionModel


fun QuestionModel.toDto(): QuestionDto {
    return QuestionDto(
        id = id,
        createdAt = createdAt,
        updatedAt = updatedAt,
        text = text,
        mediaType = mediaType,
        answerMode = answerMode,
        answers = answers.map { it.toDto() }
    )
}


fun AnswerModel.toDto(): AnswerDto {
    return AnswerDto(
        id = id,
        createdAt = createdAt,
        updatedAt = updatedAt,
        text = text,
        isCorrectAnswer = isCorrectAnswer
    )
}
